devito/dimension.py

```diff
@@ -388,6 +388,11 @@ def _arg_check(self, *args):
         """
         return
 
+    @property
+    def _arg_names(self):
```

Comment:
can you reuse something from superclass through ``super``?

is it worth moving (part of) this method to the superclass?